### PR TITLE
Release v2.0.0

### PR DESCRIPTION
**Pull request by skydiver**
More info: https://github.com/skydiver/october-plugin-forms/discussions/231

### CODE REVIEW

**Comment by skydiver** [comment]
Hey @mjauvin, are you interested in test this one?
Still under development but help will be very welcome

File: models/settings/fields.yaml [hunk 1]
```diff
@@ -6,13 +6,33 @@ tabs:
...
+        section_uploads:
+            label   : File uploads
+            type    : section
+            comment : Settings related to file uploads through components
+            tab     : martin.forms::lang.settings.tabs.general
+            cssClass: m-t-lg
+
+        global_allowed_files:
+            label       : Allowed file formats
+            type        : text
```
Comment (rubenvanerk) [inline models/settings/fields.yaml:32]:
```suggestion
            type        : taglist
```

File: components/partials/filepond.htm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    function isLoadingCheck() {
+        var isLoading = pond.getFiles().filter(x=>x.status !== 5).length !== 0;
+        if (isLoading) {
+            $('form [type="submit"]').attr("disabled", "disabled");
+        } else {
+            $('form [type="submit"]').removeAttr("disabled");
+        }
+    }
+</script>
```
Comment (rubenvanerk) [inline components/partials/filepond.htm:34]:
```suggestion
</script>
<style>
    .filepond--root {
        font-size: 16px;
    }
</style>
```
Font size is tiny at the moment. This fixes it, don't know if this is the right location though.

File: classes/Mails/AutoResponse.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            'date' => $this->record->created_at
+        ];
+
+        // CHECK FOR CUSTOM SUBJECT
+        if (!empty($this->properties['mail_resp_subject'])) {
+            $this->prepareCustomSubject();
+        }
+
+        // SET EMAIL PARAMETERS
+        $response = isset($this->properties['mail_resp_field']) ? $this->properties['mail_resp_field'] : null;
```
Comment (rubenvanerk) [inline classes/Mails/AutoResponse.php:40]:
```suggestion
        $response = $this->properties['mail_resp_field'] ?? null;
```

File: classes/Mails/AutoResponse.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        $to       = isset($this->post[$response]) ? $this->post[$response] : null;
```
Comment (rubenvanerk) [inline classes/Mails/AutoResponse.php:41]:
```suggestion
        $to       = $this->post[$response] ?? null;
```

File: classes/Mails/AutoResponse.php [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        $name     = isset($this->properties['mail_resp_name']) ? $this->properties['mail_resp_name'] : null;
```
Comment (rubenvanerk) [inline classes/Mails/AutoResponse.php:42]:
```suggestion
        $name     = $this->properties['mail_resp_name'] ?? null;
```

File: classes/Mails/AutoResponse.php [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        $from     = isset($this->properties['mail_resp_from']) ? $this->properties['mail_resp_from'] : null;
```
Comment (rubenvanerk) [inline classes/Mails/AutoResponse.php:43]:
```suggestion
        $from     = $this->properties['mail_resp_from'] ?? null;
```

File: classes/Mails/AutoResponse.php [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        $subject  = isset($this->properties['mail_resp_subject']) ? $this->properties['mail_resp_subject'] : null;
```
Comment (rubenvanerk) [inline classes/Mails/AutoResponse.php:44]:
```suggestion
        $subject  = $this->properties['mail_resp_subject'] ?? null;
```

**Review by rubenvanerk — COMMENTED** [review]
Nice work! Seems to work well. I have some small suggestions:

- [ ] When uploading I'm getting this error: `ReflectionException: Class Martin\Forms\Classes\FilePond\FilePondController does not exist`. On my machine, this can be solved by changing the `FilePond` directory in `classes` to lowercase (filepond). Same goes for `classes/Mails` directory
- [ ] Errors are not displayed in filepond

**Comment by skydiver** [comment]
Not planning to continue working with October :(

**Comment by rubenvanerk** [comment]
Are you porting this plugin to Winter or will you stop maintaining this plugin?

**Comment by skydiver** [comment]
@rubenvanerk yes, version 2 will be release only for Winter

**Pull request closed without merging**